engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/SimpleQueryFlagsStep.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.predicate.dsl;
+
+import java.util.EnumSet;
+
+/**
+ * The step in a flags definition where the simple query can be set.
+ *
+ * @param <S> The "self" type (the actual exposed type of this step)
+ */
+public interface SimpleQueryFlagsStep<S extends SimpleQueryStringPredicateOptionsStep<?>> extends SimpleQueryStringPredicateOptionsStep<S> {
+
+	/**
+	 * Enable all operation in the given flags.
+	 *
+	 * @return {@code this}, for method chaining.
+	 */
+	default SimpleQueryFlagsStep<S> all() {
```

Comment:
This is the default. Wouldn't `none` be more useful?

Reply:
It is possible to disable and enable all at once. Simply for technical correctness. By default, if null is set to all enabled according to the previous action. The inverse function that turns disable all is "none()".

If the function "disable" is used for the first time, it will set all to enabled and the indicated one to disable.

If the function "enable" is used for the first time, it will set all to disabled and the indicated one to enable.

Reply:
Myślę, że powinna być set(), która umożliwi nadpisanie ustawień. Enable i disable możliwość podania wielu parametrów:  isable(SimpleQueryFlag... operation), enable(SimpleQueryFlag... operation); This should give you a lot of options, and make the code easier.